app/main/lib/image_similarity.py

```diff
@@ -47,11 +49,30 @@ def save(image):
     db.session.rollback()
     raise e
 
+
+@tenacity.retry(wait=tenacity.wait_fixed(0.5), stop=tenacity.stop_after_delay(5), after=_after_log)
+def alter_pdq(image):
+  try:
+    # First locate existing image and append new context
+    existing = db.session.query(ImageModel).filter(ImageModel.url==image.url).one()
+    existing.pdq = image.pdq
+    flag_modified(existing, 'pdq')
+    db.session.commit()
+  except NoResultFound as e:
+    # Otherwise, add new image, but with context as an array
+    if image.context:
+      image.context = [image.context]
+    db.session.add(image)
```

Comment:
We should log the case that we fail to find the image, but we should not be (re-)adding as this will potentially create duplicates.